src/navigation.fs

```diff
@@ -74,24 +64,53 @@ module Program =
                     lastLocation <- Some window.location.href
                     Change window.location |> dispatch
                 |> box
-                    
-            window.addEventListener_popstate(unbox onChange)
-            window.addEventListener_hashchange(unbox onChange)
-            window.addEventListener(Navigation.NavigatedEvent, unbox onChange)
-        
-        let subs model =
-            Cmd.batch
-              [ [locationChanges]
-                program.subscribe model |> Cmd.map UserMsg ]
-        
-        let init () = 
-            program.init (parser window.location) |> map
-        
-        { init = init 
-          update = update
-          subscribe = subs
-          onError = program.onError
-          setState = fun model dispatch -> program.setState model (UserMsg >> dispatch) 
-          view = fun model dispatch -> program.view model (UserMsg >> dispatch) }
-    
-  
\ No newline at end of file
+
+            onChangeRef <- onChange
+
+            window.addEventListener_popstate(unbox onChangeRef)
+            window.addEventListener_hashchange(unbox onChangeRef)
+            window.addEventListener(Navigation.NavigatedEvent, unbox onChangeRef)
+
+        let unsubscribe _ =
```

Comment:
since it's a public function it should have the arg typed explicitly (even thought we call it Internal). Propose:
```fsharp
let unsubscribe () =
```
instead

Reply:
Ah yes, good catch :)